packages/flutter/lib/src/widgets/animated_scroll_view.dart

```diff
@@ -85,6 +85,92 @@ class AnimatedList extends _AnimatedScrollView {
     super.clipBehavior = Clip.hardEdge,
   }) : assert(initialItemCount >= 0);
 
+  /// A scrolling container that animates items with separators when they are inserted or removed.
+  ///
+  /// This widget's [AnimatedListState] can be used to dynamically insert or
+  /// remove items. To refer to the [AnimatedListState] either provide a
+  /// [GlobalKey] or use the static [of] method from an item's input callback.
+  ///
+  /// This widget is similar to one created by [ListView.separated].
+  ///
+  /// {@tool dartpad}
+  /// This sample application uses an [AnimatedList.separated] to create an effect when
+  /// items are removed or added to the list.
+  ///
+  /// ** See code in examples/api/lib/widgets/animated_list/animated_list_separated.0.dart **
+  /// {@end-tool}
+  ///
+  /// By default, [AnimatedList.separated] will automatically pad the limits of the
+  /// list's scrollable to avoid partial obstructions indicated by
+  /// [MediaQuery]'s padding. To avoid this behavior, override with a
+  /// zero [padding] property.
+  ///
+  /// {@tool snippet}
+  /// The following example demonstrates how to override the default top and
+  /// bottom padding using [MediaQuery.removePadding].
+  ///
+  /// ```dart
+  /// Widget myWidget(BuildContext context) {
+  ///   return MediaQuery.removePadding(
+  ///     context: context,
+  ///     removeTop: true,
+  ///     removeBottom: true,
+  ///     child: AnimatedList.separated(
+  ///       initialItemCount: 50,
+  ///       itemBuilder: (BuildContext context, int index, Animation<double> animation) {
+  ///         return Card(
+  ///           color: Colors.amber,
+  ///           child: Center(child: Text('$index')),
+  ///         );
+  ///       },
+  ///       separatorBuilder: (BuildContext context, int index, Animation<double> animation) {
+  ///         return const Divider();
+  ///       },
+  ///       removedSeparatorBuilder: (BuildContext context, int index, Animation<double> animation) {
+  ///         return const Divider();
+  ///       }
+  ///     ),
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
+  ///
+  /// See also:
+  ///
+  ///  * [SliverAnimatedList], a sliver that animates items when they are inserted
+  ///    or removed from a list.
+  ///  * [SliverAnimatedGrid], a sliver which animates items when they are
+  ///    inserted or removed from a grid.
+  ///  * [AnimatedGrid], a non-sliver scrolling container that animates items when
+  ///    they are inserted or removed in a grid.
+  ///  * [AnimatedList], which animates items added and removed from a list instead
+  ///    of a grid.
+  AnimatedList.separated({
+    super.key,
+    required AnimatedItemBuilder itemBuilder,
+    required AnimatedItemBuilder separatorBuilder,
+    required super.removedSeparatorBuilder,
```

Comment:
```suggestion
    required AnimatedItemBuilder super.removedSeparatorBuilder,
```

If someone puts `removedSeparatorBuilder: null` in this constructor, it might cause problems. Luckily, we can specify a type for the super parameter!

Reply:
that's a great catch! Applied the change :)